### PR TITLE
add 'Updating "recommended"'

### PR DESCRIPTION
**Pull request by athornton**

### CODE REVIEW

File: docs/ops/cachemachine/updating_recommended.rst [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+   }
+
+(replace the tag and description with the correct ones, obviously).
+
+If there already is such a ``SimpleRepoMan`` entry, but it has an older tag or name, replace those with the correct versions.
+
+Then synchronize cachemachine (using ArgoCD) in the correct environment.
+The deployment will automatically restart, and that will kick off any required pulls.
+Since these pulls will just be pulling "recommended" under a different name, the image will almost certainly already be cached, and therefore the pull will be near-instant.
+After a minute of runtime plus the termination time for each pod, the nublado2 options form will again show the correct data.
```
Comment (SimonKrughoff) [inline docs/ops/cachemachine/updating_recommended.rst:59]: I don't understand what this last sentence implies.  Which pod needs to terminate?

**Review by SimonKrughoff — COMMENTED** [review]
I think there are some things that need addressing before a merge can happen

File: docs/ops/cachemachine/updating_recommended.rst [hunk 2]
```diff
@@ -0,0 +1,59 @@
+######################
+Updating "recommended"
+######################
+
+The "recommended" tag for JupyterLab images is generally a weekly image, and generally it is not too far out of date.
```
[the remark this answers is not in the document]
Comment (athornton) [reply docs/ops/cachemachine/updating_recommended.rst]: Done.

File: docs/ops/cachemachine/updating_recommended.rst [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+Tagging a new container version
+--------------------------------
+
+When a new version is to be blessed, the "recommended" tag must be updated to point to the new version.
```
[the remark this answers is not in the document]
Comment (athornton) [reply docs/ops/cachemachine/updating_recommended.rst]: Done.

File: docs/ops/cachemachine/updating_recommended.rst [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+This is, sadly, necessary--there is no way to tag an image on Docker Hub without pulling and re-pushing it.
+However, the push will be a no-op, since all the layers are, by definition, already there, so while the pull may be slow, the push will be fast.
+
+The procedure is as follows:
+
+.. code-block:: sh
+
+   docker pull registry.hub.docker.com/lsstsqre/sciplat-lab:w_2021_33  # or whatever tag
+   docker tag registry.hub.docker.com/lsstsqre/sciplat-lab:w_2021_33 registry.hub.docker.com/lsstsqre:recommended
+   docker login  # This may require interaction, depending on how you've set up your docker credentials
```
[the remark this answers is not in the document]
Comment (athornton) [reply docs/ops/cachemachine/updating_recommended.rst:27]: Done.

File: docs/ops/cachemachine/updating_recommended.rst [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+However, in most environments, cachemachine only ensures pulling of the latest two weekly images, and it is not at all unusual for more than two weeks to go by before blessing a new version.
+
+Usually this doesn't matter: the image cache on a node is Least Recently Used, and the great majority of users spawn "recommended," so it's not going to be purged.
+However, there is a display bug in the nublado2 options form that can occur.
+
+If a new node has come online after the recommended weekly has rolled out of the weekly list, then, although the new node will pre-pull "recommended", it will not pre-pull the corresponding weekly by the weekly tag, and thus cachemachine, and therefore the options form, will fail to resolve "recommended" to a particular weekly, which means the description in parentheses after the image name will be empty.
+
+Fortunately, this is easy to fix.
+
+In cachemachine's ``values.yaml`` file for the affected environment, go towards the bottom and look in ``repomen``.
```
[the remark this answers is not in the document]
Comment (athornton) [reply docs/ops/cachemachine/updating_recommended.rst:45]: Done.

File: docs/ops/cachemachine/updating_recommended.rst [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+   docker pull registry.hub.docker.com/lsstsqre/sciplat-lab:w_2021_33  # or whatever tag
+   docker tag registry.hub.docker.com/lsstsqre/sciplat-lab:w_2021_33 registry.hub.docker.com/lsstsqre:recommended
+   docker login  # This may require interaction, depending on how you've set up your docker credentials
+   docker push registry.hub.docker.com/lsstsqre/sciplat-lab:recommended
+
+Updating Phalanx to ensure the "recommended" target is pre-pulled
+-----------------------------------------------------------------
+
+However, in most environments, cachemachine only ensures pulling of the latest two weekly images, and it is not at all unusual for more than two weeks to go by before blessing a new version.
```
[the remark this answers is not in the document]
Comment (athornton) [reply docs/ops/cachemachine/updating_recommended.rst]: Done.

File: docs/ops/cachemachine/updating_recommended.rst (hunk 1 above)
Comment (athornton) [reply docs/ops/cachemachine/updating_recommended.rst]: The one that is run from the pre-pulled image.

File: docs/ops/cachemachine/updating_recommended.rst [hunk 7]
```diff
@@ -0,0 +1,79 @@
+######################
+Updating "recommended"
+######################
+
+The "recommended" tag for JupyterLab images is usually a recent weekly image.
+The image marked "recommended" is guaranteed by SQuaRE to be compatible with other services and materials--such as tutorial or system testing notebooks--that we make available on RSP deployments.
+Because this process requires quite a bit of checking and sign-off from multiple stakeholders, it is possible that approving a new version for "recommended" may take more than the two weeks (for most deployments) it takes for a weekly image to roll off the default list of images to pull.
+This can cause the RSP JupyterHub options form to display empty parentheses rather than the correct target version when a user requests a lab container.
+
+This document explains how to circumvent that display bug by changing cachemachine's ``values.html`` for the appropriate instance when moving the "recommended" tag.
```
Comment (SimonKrughoff) [inline docs/ops/cachemachine/updating_recommended.rst:10]: values.yaml?

**Review by SimonKrughoff — APPROVED** [review]
Looks great.  Thanks for those changes.

File: docs/ops/cachemachine/updating_recommended.rst (hunk 7 above)
Comment (athornton) [reply docs/ops/cachemachine/updating_recommended.rst:10]: derp, yeah.

**Pull request merged**